src/meta/src/manager/catalog/mod.rs

```diff
@@ -3253,53 +3208,74 @@ impl CatalogManager {
             .in_progress_creation_streaming_job
             .remove(&subscription.id);
 
-        subscription.stream_job_status = PbStreamJobStatus::Created.into();
+        subscription.subscription_state = PbSubscriptionState::Create.into();
         subscriptions.insert(subscription.id, subscription.clone());
-        for table in &mut internal_tables {
-            table.stream_job_status = PbStreamJobStatus::Created.into();
-            tables.insert(table.id, table.clone());
-        }
-        commit_meta!(self, subscriptions, tables)?;
+        commit_meta!(self, subscriptions)?;
+        Ok(())
+    }
+
+    pub async fn notify_create_subscription(
+        &self,
+        subscription_id: SubscriptionId,
+    ) -> MetaResult<NotificationVersion> {
+        let core = &mut *self.core.lock().await;
+        let database_core = &mut core.database;
+        let subscriptions = BTreeMapTransaction::new(&mut database_core.subscriptions);
+        let subscription = subscriptions
+            .get(&subscription_id)
+            .ok_or_else(|| anyhow!("subscription not found"))?
+            .clone();
+        assert_eq!(
+            subscription.subscription_state,
+            Into::<i32>::into(PbSubscriptionState::Create)
+        );
+        commit_meta!(self, subscriptions)?;
 
         let version = self
             .notify_frontend(
                 Operation::Add,
                 Info::RelationGroup(RelationGroup {
                     relations: vec![Relation {
                         relation_info: RelationInfo::Subscription(subscription.to_owned()).into(),
-                    }]
-                    .into_iter()
-                    .chain(internal_tables.into_iter().map(|internal_table| Relation {
-                        relation_info: RelationInfo::Table(internal_table).into(),
-                    }))
-                    .collect_vec(),
+                    }],
                 }),
             )
             .await;
-
         Ok(version)
     }
 
-    pub async fn cancel_create_subscription_procedure(&self, subscription: &Subscription) {
+    pub async fn clean_dirty_subscription(&self) -> MetaResult<()> {
         let core = &mut *self.core.lock().await;
         let database_core = &mut core.database;
+        let mut subscriptions = BTreeMapTransaction::new(&mut database_core.subscriptions);
+        let mut remove_subscriptions = vec![];
+        subscriptions.tree_mut().retain(|_, s| {
```

Comment:
Using `tree_mut` will only modify the in memory btreemap, but will not store the operation on the btreemap, and therefore the later `commit_meta` will not modify the subscription metadata. 

We should first use `tree_ref` to find out the subscriptions to be removed, and then call `remove` on them one by one.